gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/CookieSessionHandler.java

```diff
@@ -121,10 +122,26 @@ public void handle(RoutingContext context) {
         }
 
         // Need to wait the session to be ready before invoking next.
-        sessionObs
-                .doOnError(t -> logger.warn("Unable to restore the session", t))
-                .doFinally(context::next)
-                .subscribe();
+        try {
+            sessionObs
+                    .doOnError(t -> {
+                        final Throwable cause = t.getCause();
+                        if (cause instanceof SignatureException) {
+                            logger.warn("Unable to restore the session: {}", cause.getMessage(), t);
+                        } else {
+                            logger.info("Unable to restore the session: {}. {}", cause.getMessage(), t.getMessage());
+                            logger.debug("Unable to restore the session: ", t);
+                        }
+                    })
+                    .doFinally(context::next)
+                    .subscribe(
+                            success -> {
+                            }, error -> {
+                            }
+                    );
+        } catch (Exception ex) {
+            logger.error("Some exception reported");
+        }
```

Comment:
**suggestion**: all unexpected exception should be logged at WARN, we should restrict the INFO level only for expired or prematured token IMO.

```suggestion
        sessionObs
                .doFinally(context::next)
                .subscribe(
                   success -> logger.trace("Session restored successfully"),
                   error -> {
                        final Throwable cause = error.getCause();
                        if (cause instanceof PrematureJWTException | error instanceof ExpiredJWTException) {
                            logger.info("Unable to restore the session: {}", cause.getMessage());
                        } else {                            
                          logger.warn("Unable to restore the session: {}", cause.getMessage());
                        }
                      }   
                );
```

Reply:
updated as suggested